tools/bin/commands/util/operator_funcs

```diff
@@ -20,25 +37,52 @@ dep_ensure() {
         dep ensure -vendor-only -v
     else
         echo "Running 'dep ensure' with $GO_BUILDER_IMAGE"
-        [ -d "dep-cache" ] || mkdir dep-cache
+        required_dirs
         local extra_opts=""
         if [ -n "$as_user" ]; then
             extra_opts="$extra_opts -u $as_user"
         fi
         docker run -w /gopath/src/github.com/syndesisio/syndesis/install/operator \
+                   -u $(id -u):$(id -g) \
                    $extra_opts \
+                   -v "$DOCKER_GOPATH_VOLUME":/gopath:z \
+                   -v "$DOCKER_CACHE_VOLUME":/.cache:z \
                    -v $operator_dir/dep-cache:/gopath/pkg/dep:z \
                    -v $operator_dir:/gopath/src/github.com/syndesisio/syndesis/install/operator:z \
                    $GO_BUILDER_IMAGE \
                    dep ensure -vendor-only -v
     fi
 }
 
+required_dirs() {
+    require_dir "$operator_dir/dep-cache"
+    require_dir "$DOCKER_CACHE_VOLUME"
```

Comment:
in the scripts we should avoid global variables as much as possible. Typically they are only used for constants, so would it be an issue to pass them via arguments around ?